script.js

```diff
@@ -0,0 +1,107 @@
+class BookShop {
+  discountRateByBookNumber ;
+  topRatedBooks;
+  setDiscountRateByBookNumber (promotions) {
+    this.discountRateByBookNumber = {...promotions}
+  }
+
+  setTopRatedBooks (books){
+    this.topRatedBooks = books
+  }
+
+  priceSummation (data) {
+    let result = []
+    let transaction = {...data}
+
+    while (Object.keys(transaction).length !== 0 ) {
```

Comment:
just a note, it is more common in js to using map

Reply:
I will try refactor this using map as recommended krub :)